### PR TITLE
Update the Taupage base image to Ubuntu 16.04 LTS

### PR DESCRIPTION
**Pull request by arjunrn**
Upgrade the base AMI to Ubuntu 16.04

Changelog:

* Updated the `cloud-init` patch to work with the library based in Ubuntu 16.04
* Install `upstart` and purge `systemd`
* Updated OpenJDK on the base AMI to Java 8
* Removed `serverspec` tests which assume that Ubuntu 16.04 has systemd.

Build changes:
* Added tags so that build and test machines/volumes can be bulk deleted. If there are test machines which are left running this causes new tests to fail. The only way to run them reliability is to ensure that only set of tests is running at a time.
* Rebuilt the docker images used for smoke tests so that they rely on the correct base image libraries.

### CODE REVIEW

**Comment by vwiessner** [comment]
:+1:

**Comment by femueller** [comment]
👍

File: build/setup.d/10-upgrade-base.sh [hunk 1]
```diff
@@ -5,6 +5,7 @@ set -x
 echo "Updating system..."
 
 # sudo rm -rf /var/lib/apt/lists/*
+chmod 777 /tmp
```
Comment (aermakov-zalando) [inline build/setup.d/10-upgrade-base.sh]: Why?

**Comment by aermakov-zalando** [comment]
LGTM, other than the permission change on `/tmp`. Is it really needed?

**Comment by arjunrn** [comment]
> LGTM, other than the permission change on `/tmp`. Is it really needed?

For some reason after the reboot after installing `upstart` the permissions on the `/tmp` directory are:

```
drwxr-xr-x   3 root root  4096 Mar 25 14:27 tmp/
```
This is the [bug](https://bugs.launchpad.net/ubuntu/+source/apt/+bug/1577926)

**Comment by aermakov-zalando** [comment]
I see. Could you set it to `1777` then?

**Comment by aermakov-zalando** [comment]
👍

**Comment by arjunrn** [comment]
:+1:

**Pull request merged**